domain.go

```diff
@@ -159,6 +159,36 @@ type DomainVideo struct {
 	Model DomainVideoModel `xml:"model"`
 }
 
+type DomainChannelSource struct {
+	Mode string `xml:"mode,attr"`
+	Path string `xml:"path,attr"`
+}
+
+type DomainChannelTarget struct {
+	Type  string `xml:"type,attr"`
+	Name  string `xml:"name,attr"`
+	State string `xml:"state,attr,omitempty"` // is guest agent connected?
+}
+
+type DomainChannelAlias struct {
+	Name string `xml:"name,attr"`
+}
+
+type DomainChannelAddress struct {
+	Type       string `xml:"type,attr"`
+	Controller int    `xml:"controller,attr"`
+	Bus        int    `xml:"bus,attr"`
+	Port       int    `xml:"port,attr"`
```

Comment:
Most/all of these fields need to be optional depending on what "Type" refers to. Unfortunately we can't use "omitempty" for this purpose, since 0 is a valid value for most of them. So reluctantly I think they need to be strings with omitempty